lib/api/objectGet.js

```diff
@@ -80,68 +78,51 @@ function objectGet(authInfo, request, log, callback) {
         const responseMetaHeaders = collectResponseHeaders(objMD,
             corsHeaders, verCfg);
 
-        // 0 bytes file
-        if (objMD.location === null) {
-            const range = request.headers.range;
-            // aws returns the empty object if there is an invalid range but
-            // sends an invalidRange error if the range is valid on an empty
-            // object
-            if (range && validRangeOnEmptyFile(range)) {
-                return callback(errors.InvalidRange, null, corsHeaders);
-            }
-            pushMetric('getObject', log, {
-                authInfo,
-                bucket: bucketName,
-                newByteLength: 0,
-            });
-            return callback(null, null, responseMetaHeaders);
-        }
-        let range;
-        let maxContentLength;
+        const objLength = (objMD.location === null ?
+                           0 : parseInt(objMD['content-length'], 10));
+        let byteRange;
         if (request.headers.range) {
-            maxContentLength =
-              parseInt(responseMetaHeaders['Content-Length'], 10);
-            responseMetaHeaders['Accept-Ranges'] = 'bytes';
-            const parseRangeRes = parseRange(request.headers.range,
-              maxContentLength);
-            range = parseRangeRes.range;
-            const error = parseRangeRes.error;
+            const { range, error } = parseRange(request.headers.range,
+                                                objLength);
             if (error) {
                 return callback(error, null, corsHeaders);
             }
+            responseMetaHeaders['Accept-Ranges'] = 'bytes';
             if (range) {
-                // End of range should be included so + 1
+                byteRange = range;
                 responseMetaHeaders['Content-Length'] =
-                    Math.min(maxContentLength - range[0],
-                    range[1] - range[0] + 1);
-                responseMetaHeaders['Content-Range'] = `bytes ${range[0]}-`
-                    + `${Math.min(maxContentLength - 1, range[1])}` +
-                    `/${maxContentLength}`;
+                    range[1] - range[0] + 1;
```

Comment:
I removed it because I though that the additional info given was not worth the additional like taken in the code (as you may have guessed :p) but will add it back then if you think +1 is mysterious :)